server/prefabs/Enemy.cpp

```diff
@@ -1,9 +1,52 @@
 #include "prefabs/Enemy.hpp"
 #include "NetTransform.hpp"
+#include "CollisionManager.hpp"
+#include "Player.hpp"
+//#include "Health.hpp"
 
-float s = 0;
+Enemy::Enemy() : Entity() {
+    Collider* hitbox = new Collider(this);
+    //TODO: test size values
+    hitbox->SetRadius(50);
+    hitbox->SetHeight(20);
+
+    AddComponent(hitbox);
+    CollisionManager::instance().add(this);
+    this->currentPhase = PHASE1;
+}
+
+Enemy::Enemy(int networkId) : Entity(networkId){
+    Collider* hitbox = new Collider(this);
+    //TODO: test size values
+    hitbox->SetRadius(50);
+    hitbox->SetHeight(20);
+    
+    AddComponent(hitbox); // TODO: decrement player health if they hit the boss
+    CollisionManager::instance().add(this);
+    this->currentPhase = PHASE1;
+}
+
+float s = 0; // time in milliseconds
 void Enemy::update(float deltaTime) {
     s += deltaTime;
-    // position += glm::vec3(glm::sin(s), 0, 0);
+
+    // moves in a circle
+    // TODO: use colliderManager.move instead
     GetComponent<NetTransform>()->position += glm::vec3(glm::sin(s), 0, 0);
+
+    // every 5 seconds, attack
+    if(std::fmod(s, 5000.0) == 0.0){
```

Comment:
Not sure if this is tested, but I think we may need to consider what happens when s does not meet exactly multiples of 5000 ms. For example, if we are updating the server at 30ms ticks like professor described in first class, s will miss 5000ms but instead be 4080ms, 5010ms, ... I think we can consider something like std::fmod(s, 5000.0) < deltatime, or some other logic that makes the boss not miss an attack.